In [1]:
import pandas as pd
import numpy as np
import seaborn as sns 
import matplotlib as plt

In [3]:
sdata=pd.read_csv('diabetes.csv')

In [5]:
sdata


Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0              6      148             72             35        0  33.6   
1              1       85             66             29        0  26.6   
2              8      183             64              0        0  23.3   
3              1       89             66             23       94  28.1   
4              0      137             40             35      168  43.1   
..           ...      ...            ...            ...      ...   ...   
763           10      101             76             48      180  32.9   
764            2      122             70             27        0  36.8   
765            5      121             72             23      112  26.2   
766            1      126             60              0        0  30.1   
767            1       93             70             31        0  30.4   

     DiabetesPedigreeFunction  Age  Outcome  
0                       0.627   50        1  
1                       0.351   31        0  
2                       0.672   32        1  
3                       0.167   21        0  
4                       2.288   33        1  
..                        ...  ...      ...  
763                     0.171   63        0  
764                     0.340   27        0  
765                     0.245   30        0  
766                     0.349   47        1  
767                     0.315   23        0  

[768 rows x 9 columns]

In [7]:
newdata=sdata.Outcome>0

In [13]:
newdataF=sdata[newdata]

In [15]:
newdataF

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0              6      148             72             35        0  33.6   
2              8      183             64              0        0  23.3   
4              0      137             40             35      168  43.1   
6              3       78             50             32       88  31.0   
8              2      197             70             45      543  30.5   
..           ...      ...            ...            ...      ...   ...   
755            1      128             88             39      110  36.5   
757            0      123             72              0        0  36.3   
759            6      190             92              0        0  35.5   
761            9      170             74             31        0  44.0   
766            1      126             60              0        0  30.1   

     DiabetesPedigreeFunction  Age  Outcome  
0                       0.627   50        1  
2                       0.672   32        1  
4                       2.288   33        1  
6                       0.248   26        1  
8                       0.158   53        1  
..                        ...  ...      ...  
755                     1.057   37        1  
757                     0.258   52        1  
759                     0.278   66        1  
761                     0.403   43        1  
766                     0.349   47        1  

[268 rows x 9 columns]

In [32]:
meddata=newdataF.Age<50

In [30]:
meddataF=newdataF[meddata]
meddataF

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0              6      148             72             35        0  33.6   
2              8      183             64              0        0  23.3   
4              0      137             40             35      168  43.1   
6              3       78             50             32       88  31.0   
8              2      197             70             45      543  30.5   
..           ...      ...            ...            ...      ...   ...   
755            1      128             88             39      110  36.5   
757            0      123             72              0        0  36.3   
759            6      190             92              0        0  35.5   
761            9      170             74             31        0  44.0   
766            1      126             60              0        0  30.1   

     DiabetesPedigreeFunction  Age  Outcome  
0                       0.627   50        1  
2                       0.672   32        1  
4                       2.288   33        1  
6                       0.248   26        1  
8                       0.158   53        1  
..                        ...  ...      ...  
755                     1.057   37        1  
757                     0.258   52        1  
759                     0.278   66        1  
761                     0.403   43        1  
766                     0.349   47        1  

[268 rows x 9 columns]

In [34]:
middledata=meddataF.Age>30
middledataF=meddataF[middledata]
middledataF

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0              6      148             72             35        0  33.6   
2              8      183             64              0        0  23.3   
4              0      137             40             35      168  43.1   
8              2      197             70             45      543  30.5   
9              8      125             96              0        0   0.0   
..           ...      ...            ...            ...      ...   ...   
755            1      128             88             39      110  36.5   
757            0      123             72              0        0  36.3   
759            6      190             92              0        0  35.5   
761            9      170             74             31        0  44.0   
766            1      126             60              0        0  30.1   

     DiabetesPedigreeFunction  Age  Outcome  
0                       0.627   50        1  
2                       0.672   32        1  
4                       2.288   33        1  
8                       0.158   53        1  
9                       0.232   54        1  
..                        ...  ...      ...  
755                     1.057   37        1  
757                     0.258   52        1  
759                     0.278   66        1  
761                     0.403   43        1  
766                     0.349   47        1  

[178 rows x 9 columns]

In [38]:
middledata.sum()

178

there are 178 people who are diabetes +ve and and there age is between 30 to 50 year

In [45]:
agedata=sdata.Age>30

In [47]:
agedataF=sdata[agedata]

In [49]:
agedataF

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0              6      148             72             35        0  33.6   
1              1       85             66             29        0  26.6   
2              8      183             64              0        0  23.3   
4              0      137             40             35      168  43.1   
8              2      197             70             45      543  30.5   
..           ...      ...            ...            ...      ...   ...   
759            6      190             92              0        0  35.5   
761            9      170             74             31        0  44.0   
762            9       89             62              0        0  22.5   
763           10      101             76             48      180  32.9   
766            1      126             60              0        0  30.1   

     DiabetesPedigreeFunction  Age  Outcome  
0                       0.627   50        1  
1                       0.351   31        0  
2                       0.672   32        1  
4                       2.288   33        1  
8                       0.158   53        1  
..                        ...  ...      ...  
759                     0.278   66        1  
761                     0.403   43        1  
762                     0.142   33        0  
763                     0.171   63        0  
766                     0.349   47        1  

[351 rows x 9 columns]

In [55]:
midage=agedataF.Age<50
midage.sum()

262

**here are total 262 peoples whose age is between 30 year to 50 year**

**NOW WE WILL calculate the percentage in this age**

In [61]:
percent=((middledata.sum())/midage.sum())*100

In [63]:
percent

67.93893129770993

**The per centage of Being Diabetes positive in people age between 30 to 50 year is ****68%**

**The CONCLUSION**

1.**the diabeties +ve percentage in old persons (age>50)** is **47%**

2.**the diabeties +ve percentage in middle age persons (30< age <50)** is **68%**

3.**the diabeties +ve percentage in Adults and Teenegers persons (age<30)** is **21%**

***FINAL***

**IF your age is between 30 to 50 years be careful of being DIABETES +ve**